src/AppBundle/Repository/EventRepository.php

```diff
@@ -51,26 +53,85 @@ public function getNumNewEditors(Event $event)
-    //         ->andwhere('rev_parent_id = 0')
-    //         ->andwhere('rev_timestamp BETWEEN :start AND :end')
-    //         ->andwhere('rev_user_text IN ("MusikAnimal")')
-    //         ->setParameter('start', $start)
-    //         ->setParameter('end', $end);
-    //     $stmt = $rqb->execute();
-
-    //     return $stmt->fetchColumn(0);
-    // }
+    /**
+     * Get the database names of the EventWiki's belonging to the Event.
+     * @param  Event $event
+     * @return string[]
+     */
+    public function getDbNames(Event $event)
+    {
+        $projectUrls = array_map(function ($eventWiki) {
+            return 'https://'.$eventWiki->getDomain().'.org';
+        }, $event->getWikis()->toArray());
+
+        $conn = $this->getMetaConnection();
+        $rqb = $conn->createQueryBuilder();
+        $rqb->select(["CONCAT(dbname, '_p') AS dbname"])
+            ->from('wiki')
+            ->where('url IN (:projectUrls)')
+            ->setParameter('projectUrls', $projectUrls, Connection::PARAM_STR_ARRAY);
+        $stmt = $rqb->execute();
+        return array_column($stmt->fetchAll(), 'dbname');
+    }
+
+    /**
+     * Get the number of pages created within the timeframe and for the given users.
+     * @param  string   $dbName Database name such as 'enwiki_p'.
+     * @param  DateTime $start
+     * @param  DateTime $end
+     * @param  string[] $usernames
+     * @return int
+     */
+    public function getNumPagesCreated($dbName, DateTime $start, DateTime $end, $usernames)
+    {
+        $start = $start->format('YmdHis');
+        $end = $end->format('YmdHis');
+
+        $conn = $this->getReplicaConnection();
+        $rqb = $conn->createQueryBuilder();
+
+        $rqb->select('COUNT(page_title)')
+            ->from("$dbName.page")
+            ->join("$dbName.page", "$dbName.revision_userindex", null, 'rev_page = page_id')
+            ->where('page_namespace = 0')
+            ->andwhere('rev_parent_id = 0')
+            ->andwhere('rev_timestamp BETWEEN :start AND :end')
+            ->andwhere('rev_user_text IN (:usernames)')
+            ->setParameter('start', $start)
+            ->setParameter('end', $end)
+            ->setParameter('usernames', $usernames, Connection::PARAM_STR_ARRAY);
+        $stmt = $rqb->execute();
+
+        return $stmt->fetchColumn(0);
+    }
+
+    /**
+     * Get the number of pages edited within the timeframe and for the given users.
+     * @param  string   $dbName Database name such as 'enwiki_p'.
+     * @param  DateTime $start
+     * @param  DateTime $end
+     * @param  string[] $usernames
+     * @return int
+     */
+    public function getNumPagesEdited($dbName, DateTime $start, DateTime $end, $usernames)
+    {
+        $start = $start->format('YmdHis');
+        $end = $end->format('YmdHis');
+
+        $conn = $this->getReplicaConnection();
+        $rqb = $conn->createQueryBuilder();
+
+        $rqb->select('COUNT(page_title)')
```

Comment:
COUNT(DISTINCT page_title)

Reply:
Since the only difference between the Created and Edited queries is that we check for rev_parent_id = 0 in Edited and not in Created - could we make use of the Edited result set to get the Created stats? Do you think it could be a performance improvement?

Reply:
Good catch on the `DISTINCT`! And yeah, we might be able to combine the Created and Edited queries somehow. I've got something in mind that might work.